perceval/converters/circuit_to_graph_converter.py

```diff
@@ -0,0 +1,187 @@
+    gates_names = []
+    gates_qubits = []
+    for instr in qiskit_circuit.data:
+        # Extract gate name and qubits
+        gates_names.append(instr[0].name)
+        qubits = instr[1]
+        qubit_numbers = [qiskit_circuit.find_bit(q).index for q in qubits]
+        gates_qubits.append(qubit_numbers)
+    return gates_names, gates_qubits
+
+
+NUMBER_REPETITIONS_IF_MIN_CNOTS = 30
+
+
+class CircuitToGraphConverter:
+    """
+    Takes a qiskit circuit and converts it into a graph.
+    Initially get the list of gates and the qubits on which each gate is acting on.
+    With this output or if the user already has the two lists, it converts it into
+    a graph where the vertices are the qubits and the edges the gates acting on them,
+    the weight of its edges depends on the type of gate.
+    """
+
+    def __init__(self, qiskit_circuit: QuantumCircuit = None, gates=None, qubits=None):
+        """
+        :param qiskit_circuit: Quantum circuit to convert.
+        :type qiskit_circuit: QuantumCircuit
+        :type gates: list of strings
+        :type qubits: list of lists of ints
+        """
+        if gates is not None and qubits is not None:
+            self.gates = gates
+            self.qubits = qubits
+        elif qiskit_circuit is not None:
+            self.gates, self.qubits = gates_and_qubits(qiskit_circuit)
+        else:
+            raise ValueError("Either a Qiskit circuit or both gates and qubits lists must be provided")
+
+    def graph_generator(self) -> nx.Graph:
+        """
+        The qubits of the circuit are interpreted as nodes. The weight of the edges
+        connecting them will depend on the number and type of gates.
+        """
+        elements_set = set(item for sublist in self.qubits for item in sublist)  # nodes
+        g = nx.Graph()
+        g.add_nodes_from(elements_set)
+        edge_weights = {}
+
+        # Iterate over every gate and add edges between pairs of nodes
+        for l in range(len(self.gates)):
+            subset = sorted(self.qubits[l])
+            for i in range(len(subset)):
+                for j in range(i + 1, len(subset)):
+                    edge = (subset[i], subset[j])
+                    # Increase the weight of the edge by 1 if the gate is in the list. Increase by 2 if not.
+                    if self.gates[l] in ['cx', 'cy', 'cz', 'ch', 'cp']:
+                        if edge in edge_weights:
+                            edge_weights[edge] += 1
+                        else:
+                            edge_weights[edge] = 1
+                    else:
+                        if edge in edge_weights:
+                            edge_weights[edge] += 2
+                        else:
+                            edge_weights[edge] = 2
+
+        for edge, weight in edge_weights.items():
+            g.add_edge(*edge, weight=weight)
+        return g
+
+    @staticmethod
+    def plot_graph(g):  # g is the graph created in graph_generator
```

Comment:
Shouldn't you remove this function now that pdisplay is ready to take a graph as parameter.